### PR TITLE
Update 99problems.md

### PR DESCRIPTION
**Pull request by LambdaEnthusiast**
Proposing a more direct implementation of (exact k list) which also prunes the search and outputs the choices in a more canonical order. The original is a bit hard to follow for me, so I'm not sure if it prunes or not. The original seems to generate a lot of heap garbage to me but maybe OCaml knows how to optimize that out.

### CODE REVIEW

**Comment by lefessan** [comment]
Why use an auxiliary function when the problem is directly recursive:

```
let rec extract k list =
  if k <= 0 then [ [ ] ] 
  else
    match list with
    | [ ] -> [ ]
    | head :: tail ->
      let without_head = extract k tail in
      let with_head = List.map (fun l -> head :: l) (extract (k-1) tail) in
      without_head @ with_head
;;
```

File: site/learn/tutorials/99problems.md [hunk 1]
```diff
@@ -605,6 +605,18 @@ SOLUTION
 >   aux k [] emit list;;
 > ```
 
+> ```ocamltop
+> let extract k l =
+>   let rec aux k n l p a =
+>     (if ((k < 0) || (k > n))
```
Comment (lefessan) [inline site/learn/tutorials/99problems.md:611]: two extra parens:
`if k < 0 || k > n`

File: site/learn/tutorials/99problems.md [hunk 2]
```diff
@@ -605,6 +605,18 @@ SOLUTION
...
+>      then a
+>      else
+>        (match l with
```
Comment (lefessan) [inline site/learn/tutorials/99problems.md:614]: extra paren (not needed)

File: site/learn/tutorials/99problems.md [hunk 3]
```diff
@@ -605,6 +605,18 @@ SOLUTION
...
+>        | [] -> ((List.rev p) :: a)
```
Comment (lefessan) [inline site/learn/tutorials/99problems.md:615]: two extra parens, but keeping one is better for readability:
`| [] -> (List.rev p) :: a`

File: site/learn/tutorials/99problems.md [hunk 4]
```diff
@@ -605,6 +605,18 @@ SOLUTION
...
+>        | (hd :: tl) -> (aux (k-1) (n-1) tl (hd :: p) (aux k (n-1) tl p a))))
```
Comment (lefessan) [inline site/learn/tutorials/99problems.md:616]: extra parens:
`| hd :: tl -> aux (k-1) (n-1) tl (hd :: p) (aux k (n-1) tl p a)`

File: site/learn/tutorials/99problems.md [hunk 5]
```diff
@@ -605,6 +605,18 @@ SOLUTION
...
+>   in (aux k (List.length l) l [] []);;
```
Comment (lefessan) [inline site/learn/tutorials/99problems.md:617]: extra paren:
`in aux k (List.length l) l [] [];;`

**Comment by Chris00** [comment]
Simplified the solution in fb72958.
